drivers/net/wireless/ath/ath10k/mac.c

```diff
@@ -7900,6 +7907,67 @@ struct ath10k_vif *ath10k_get_arvif(struct ath10k *ar, u32 vdev_id)
 	return arvif_iter.arvif;
 }
 
+#define ATH10K_SMBIOS_VHT_CAP_OFFSET	0x4
+#define ATH10K_SMBIOS_CTRY_FLAG_OFFSET	0x5
+#define ATH10K_SMBIOS_CTRY_CODE_OFFSET	0x6
+
+static void ath10k_mac_check_smbios(const struct dmi_header *hdr, void *data)
+{
+	struct ath10k *ar = data;
+	u8 vht_disable;
+	u8 ctry_flag;
+	u16 ctry_code;
+	u16 rd;
```

Comment:
Since you are checking later `rd & COUNTRY_ERD_FLAG` better to initialize this to `0` just to avoid reading rubbish if you return from this function before setting a proper value in `rd`